### PR TITLE
Add support for cracking Padlock databases

### PR DESCRIPTION
**Pull request by kholia**
Sample Padlock databases for testing,

* [padlock-empty-openwall.txt](https://github.com/magnumripper/JohnTheRipper/files/837456/padlock-empty-openwall.txt)
* [padlock-something-openwall.txt](https://github.com/magnumripper/JohnTheRipper/files/837458/padlock-something-openwall.txt)
* [padlock-password@12345.txt](https://github.com/magnumripper/JohnTheRipper/files/837460/padlock-password.12345.txt) (password is "password@12345")

Fixes https://github.com/magnumripper/JohnTheRipper/issues/2333.

### CODE REVIEW

File: src/padlock_fmt_plug.c [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+	MEM_FREE(saved_key);
+}
+
+static int valid(char *ciphertext, struct fmt_main *self)
+{
+	char *p = ciphertext, *ctcopy, *keeptr;
+	int extra;
+	int res;
+
+	if (strncasecmp(ciphertext, FORMAT_TAG, FORMAT_TAG_LENGTH))
```
Comment (magnumripper) [inline src/padlock_fmt_plug.c]: This should be case sensitive! You're opening a huge can of worms here.

**Comment by kholia** [comment]
Thanks for catching this. It was a copy-paste error.

**Comment by kholia** [comment]
The new commit also introduces a `memset(output, 0, 4096)` call to avoid possible false positives caused by older "valid" decrypted data.

**Comment by kholia** [comment]
This should be OK to merge now, merging!

**Comment by kholia** [comment]
The requested changes were done in the latest merged commit.

**Pull request merged**